File: clients/typescript/src/SystemdUnitChange.ts
import {SystemdUnitChangeState} from './SystemdUnitChangeState';
export class SystemdUnitChange {
  private _change: SystemdUnitChangeState;
  private _file: string;
  private _destination: string;

  constructor(input: {
    change: SystemdUnitChangeState,
    file: string,
    destination: string,
  }) {
    this._change = input.change;
    this._file = input.file;
    this._destination = input.destination;
  }

  get change(): SystemdUnitChangeState { return this._change; }
  set change(change: SystemdUnitChangeState) { this._change = change; }

  get file(): string { return this._file; }
  set file(file: string) { this._file = file; }

  get destination(): string { return this._destination; }
  set destination(destination: string) { this._destination = destination; }
}

File: clients/printnanny-os/rust/src/systemd_unit_change.rs
// SystemdUnitChange represents a SystemdUnitChange model.
#[derive(Clone, Debug, Deserialize, Eq, Hash, Ord, PartialEq, PartialOrd, Serialize)]
pub struct SystemdUnitChange {
    #[serde(rename="change")]
    pub change: Box<crate::SystemdUnitChangeState>,
    #[serde(rename="file")]
    pub file: String,
    #[serde(rename="destination")]
    pub destination: String,
}

impl SystemdUnitChange {
    pub fn new(change: crate::SystemdUnitChangeState, file: String, destination: String) -> SystemdUnitChange {
        SystemdUnitChange {
            change: Box::new(change),
            file,
            destination,
        }
    }
}
